src/datasources/entity-ds-react/queries/queryBuilder.ts

```diff
@@ -171,3 +173,41 @@ export const buildQueryFilter = (filter: API.Filter, request: EntityQueryRequest
     substitute(clonedFilter.clauses, request, templateSrv)
     return clonedFilter
 }
+
+/**
+ * Merge any filters having selected values from the FilterPanelReact into the given filter.
+ * @param entityType An EntityTypes value
+ */
+export const mergeFilterPanelFilters = (entityType: string, originalFilter: API.Filter, filterEditorData: FilterEditorData) => {
+    // only include filters targeting the given entityType
+    const filtersToMerge = filterEditorData?.activeFilters.filter(f => f.entity.label === entityType) || []
+
+    for (const filter of filtersToMerge) {
+        const filterId = getFilterId(filter)
+        const selectableValues = filterEditorData?.selectableValues.find(sv => sv.filterId === filterId)
+
+        if (selectableValues?.values) {
+            // get all selected values that are non-empty, converted to strings
+            const values = selectableValues?.values.map(sv => (sv.value || '').toString()).filter(v => v.length > 0)
+
+            if (values) {
```

Comment:
Nit: We could probably break this function down into a few chunks. 4 levels of indentation feels like a lot.